docs/decisions/0002-architecture.rst

```diff
@@ -0,0 +1,119 @@
+0002 Architecture
+#################
+
+.. TODO: This document will be moved to a plugin repo once we have a plugin architecture.
+
+Status
+******
+
+**Provisional**
+
+
+Context
+*******
+
+#. This Django app generates and shows user certificates.
+#. The models should store certificate configurations. The certificate types will vary between different course types.
+   The available course types are:
+
+   #. :ref:`Course <course>`.
+   #. :ref:`Lesson <lesson>`.
+   #. :ref:`Pathway <pathway>`.
+
+#. This Django app uses `celerybeat`_ to periodically retrieve data from the external API. Retrieving this data is
+   pluggable - it means that other developers can develop a Python package and install it to have a custom ways to
+   retrieve data from different APIs.
+#. If a user matches the criteria, the certificates will be generated from a PDF template (stored in the assets model).
+   The PDF will be uploaded to S3, and the link will be sent to the user. The generation process should also be pluggable - it means
+   that other developers can develop a Python package and install it to have custom ways to generate certificates.
+
+.. _celerybeat: https://django-celery-beat.readthedocs.io/en/latest/
+
+
+Decision
+********
+
+.. graphviz::
```

Comment:
It does not seem to be working on ReadTheDocs. When I build it locally, it looks like this:

![image](https://github.com/open-craft/openedx-certificates/assets/3189670/3ad665c9-5537-4edb-81f1-f63aa873eb30)

Reply:
Fixed in c9fb3e96d89606dec45350c36c50a85ded8b506e.